dipy/tracking/streamlinespeed.pyx

```diff
@@ -276,15 +305,44 @@ def set_number_of_points(streamlines, nb_points=3):
     >>> modified_streamline = set_number_of_points(streamline, 3)
     >>> len(modified_streamline)
     3
-    >>> # Multiple streamlines
+
+    Multiple streamlines:
+
     >>> streamlines = [streamline, streamline[::2]]
-    >>> modified_streamlines = set_number_of_points(streamlines, 10)
+    >>> new_streamlines = set_number_of_points(streamlines, 10)
     >>> [len(s) for s in streamlines]
     [100, 50]
-    >>> [len(s) for s in modified_streamlines]
+    >>> [len(s) for s in new_streamlines]
     [10, 10]
 
     '''
+    if isinstance(streamlines, Streamlines):
+        if len(streamlines) == 0:
+            return Streamlines()
+
+        nb_streamlines = len(streamlines)
+        dtype = streamlines._data.dtype
+        new_streamlines = Streamlines()
+        new_streamlines._data = np.zeros((nb_streamlines * nb_points, 3),
+                                         dtype=dtype)
+        new_streamlines._offsets = nb_points * np.arange(nb_streamlines,
+                                                         dtype=np.intp)
+        new_streamlines._lengths = nb_points * np.ones(nb_streamlines,
+                                                       dtype=np.intp)
+
+        if dtype == np.float32:
+            c_set_number_of_points_from_arraysequence[float2d](
+                streamlines._data, streamlines._offsets,
```

Comment:
It seems you are using the protected variables in many places. So, I just wonder why there is not get/set property for these variables on Nibabel?